phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1532,23 +1538,26 @@ private PTable getTableFromCells(List<Cell> tableCellList, List<List<Cell>> allC
                 long columnTimestamp =
                     columnCellList.get(0).getTimestamp() != HConstants.LATEST_TIMESTAMP ?
                         columnCellList.get(0).getTimestamp() : timeStamp;
+                boolean isSalted = saltBucketNum != null
```

Comment:
If we're doing an ALTER we need to check both the original version of the table AND the Cells from the ALTER to see if the table's salted